extensions/interactions/ImageClickInput/directives/ImageClickInput.js

```diff
@@ -70,18 +96,27 @@ oppia.directive('oppiaInteractiveImageClickInput', [
             }
           };
           $scope.onMousemoveImage = function(event) {
-            var image = $($element).find('.oppia-image-click-img');
-            $scope.mouseX = (event.pageX - image.offset().left) / image.width();
-            $scope.mouseY = (event.pageY - image.offset().top) / image.height();
-            $scope.currentlyHoveredRegions = [];
-            for (var i = 0; i < imageAndRegions.labeledRegions.length; i++) {
-              var labeledRegion = imageAndRegions.labeledRegions[i];
-              var regionArea = labeledRegion.region.area;
-              if (regionArea[0][0] <= $scope.mouseX &&
-                  $scope.mouseX <= regionArea[1][0] &&
-                  regionArea[0][1] <= $scope.mouseY &&
-                  $scope.mouseY <= regionArea[1][1]) {
-                $scope.currentlyHoveredRegions.push(labeledRegion.label);
+            $scope.$on('newCardAvailable', function(evt, data) {
```

Comment:
Why is this declared inside $scope.onMousemoveImage (and not one level outside, or in an init() function)? Presumably you want it to trigger at the outset when the directive is initialized, not when the mouse is first moved?

Reply:
This was to disable the image when he had clicked on correct region in image and some feedback was given back to him. Here, I think  directive is not initialized again.

Reply:
Yep, but can it be declared between line 97 and 98 (i.e. in the outer scope)?

Reply:
Ok, ya, that would work.

Reply:
Done